central/auth/service/service_impl_test.go

```diff
@@ -126,3 +182,464 @@ func (s *authServiceAccessControlTestSuite) TestAuthServiceResponse() {
+	}
+
+	resp, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+		Config: config,
+	})
+	s.NoError(err)
+	s.NotEmpty(resp.GetConfig().GetId())
+}
+
+func (s *authServiceAccessControlTestSuite) TestAddGenericConfigWithID() {
+	config := &v1.AuthMachineToMachineConfig{
+		Id:                      "80c053c2-24a7-4b97-bd69-85b3a511241e",
+		TokenExpirationDuration: "1h",
+		Type:                    v1.AuthMachineToMachineConfig_GENERIC,
+		Issuer:                  "https://stackrox.io",
+	}
+
+	resp, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+		Config: config,
+	})
+	s.NoError(err)
+	s.NotEmpty(resp.GetConfig().GetId())
+}
+
+func (s *authServiceAccessControlTestSuite) TestListConfigs() {
+	config1, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+		Config: &v1.AuthMachineToMachineConfig{
+			TokenExpirationDuration: "1h",
+			Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+			Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+				{
+					Key:   "sub",
+					Value: "something",
+					Role:  testRole1,
+				},
+				{
+					Key:   "aud",
+					Value: "github",
+					Role:  testRole3,
+				},
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	config2, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+		Config: &v1.AuthMachineToMachineConfig{
+			TokenExpirationDuration: "1h",
+			Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+			Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+				{
+					Key:   "sub",
+					Value: "somewhere",
+					Role:  testRole1,
+				},
+				{
+					Key:   "aud",
+					Value: "the",
+					Role:  testRole2,
+				},
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	configs, err := s.svc.ListAuthMachineToMachineConfigs(s.accessCtx, nil)
+	s.NoError(err)
+
+	s.ElementsMatch(configs.GetConfigs(), []*v1.AuthMachineToMachineConfig{config1.GetConfig(), config2.GetConfig()})
+}
+
+func (s *authServiceAccessControlTestSuite) TestUpdateExistingConfig() {
+	config, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+		Config: &v1.AuthMachineToMachineConfig{
+			TokenExpirationDuration: "1h",
+			Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+			Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+				{
+					Key:   "sub",
+					Value: "something",
+					Role:  testRole1,
+				},
+				{
+					Key:   "aud",
+					Value: "github",
+					Role:  testRole3,
+				},
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	config.GetConfig().Mappings = []*v1.AuthMachineToMachineConfig_Mapping{
+		{
+			Key:   "sub",
+			Value: "someone",
+			Role:  testRole2,
+		},
+	}
+
+	_, err = s.svc.UpdateAuthMachineToMachineConfig(s.accessCtx,
+		&v1.UpdateAuthMachineToMachineConfigRequest{Config: config.GetConfig()})
+	s.NoError(err)
+}
```

Comment:
It could be illustrative to get the config after update and compare for equality with `config`. This means: what we updated last time is what's stored in the database.

Up to you.